### ISSUE TITLE
Upgrading from terraform google plugin 1.4 to 1.5 disables deployment

### ISSUE THREAD

**Issue by hamshif** [terraform-providers/terraform-provider-google#1013]
### Terraform Version
Terraform v0.11.0
+ provider.google v1.5.0

Your version of Terraform is out of date! The latest version
is 0.11.2. You can update by downloading from www.terraform.io/downloads.html

### Affected Resource(s)
- google_container_node_pool
- google_container_cluster

It may be an issue with Terraform's core.

### Terraform Configuration Files
```hcl

resource "google_container_node_pool" "rtp_container_cluster_node_pool" {
  name = "node-pool"
  zone = "${var.zone}"
  cluster = "${google_container_cluster.rtp_container_cluster.name}"

  node_config {
    machine_type = "${var.machine_type_node}"
    disk_size_gb = "${var.disk_size_gb_node}"
    local_ssd_count = "${var.local_ssd_count_node}"
    oauth_scopes = "${var.oauth_scopes}"
  }


  # Autoscale
  autoscaling {
    min_node_count = "${var.minNodeCount_node}"
    max_node_count = "${var.maxNodeCount_node}"
  }
}



resource "google_container_cluster" "rtp_container_cluster" {
  name = "${var.container_cluster_name}-master"
  zone = "${var.zone}"
  initial_node_count = "${var.cluster_count}"
  additional_zones = "${var.additional_zones}"
  network = "${var.network}"
  subnetwork = "${var.subnetwork}"
  enable_kubernetes_alpha = "${var.enable_kubernetes_alpha}"
  min_master_version = "${var.min_master_version}"

  master_auth {
    username = "${var.username}"
    password = "${var.password}"
  }

  node_config {
    machine_type = "${var.machine_type}"
    disk_size_gb = "${var.disk_size_gb}"
    local_ssd_count = "${var.local_ssd_count}"
    oauth_scopes = "${var.oauth_scopes}"
    tags = "${var.tags}"

    labels {
      purpose = "dev-rtp-poc-cluster"
    }
  }

  addons_config {
    http_load_balancing {
      disabled = "${var.http_load_balancing_disable}"
    }

    horizontal_pod_autoscaling {
      disabled = "${var.horizontal_pod_autoscaling_disable}"
    }
  }

}
```

```hcl
variable "project_id" {
  type = "string"
  description = "Project ID."
  default = "some-gcp-poc"
}

variable "deploy_zone" {
  type = "string"
  description = "Deploy Zone."
  default = "europe-west1-b"
}


variable "json_credentials_path" {
  type = "string"
  description = "Deploy Zone."
}

provider "google" {
  credentials = "${file(var.json_credentials_path)}"
  project     = "${var.project_id}"
  region      = "${var.deploy_zone}"
}
```

```hcl
variable "container_cluster_name" {
  type = "string"
  description = "The name of the cluster, unique within the project and zone"
}

variable "enable_kubernetes_alpha" {
  type = "string"
  description = "allow cutting edge features"
}


variable "min_master_version" {
  type = "string"
  description = "kubernetes version of cluster nodes"
}



variable "zone" {
  type = "string"
  description = "The zone that the master and the number of nodes specified in initial_node_count should be created in"
}

variable "cluster_count" {
  type = "string"
  description = "The number of nodes to create in this cluster (not including the Kubernetes master)"
}

variable "additional_zones" {
  type = "list"
  description = "If additional zones are configured, the number of nodes specified in initial_node_count is created in all specified zones"
}

variable "machine_type" {
  type = "string"
  description = "The name of a Google Compute Engine machine type"
}

variable "disk_size_gb" {
  type = "string"
  description = "Size of the disk attached to each node, specified in GB"
}

variable "local_ssd_count" {
  type = "string"
  description = " The amount of local SSD disks that will be attached to each cluster node"
}

variable "oauth_scopes" {
  type = "list"
  description = "The set of Google API scopes to be made available on all of the node VMs under the default service account"
}

variable "tags" {
  type = "list"
  description = "The list of instance tags applied to all nodes. Tags are used to identify valid sources or targets for network firewalls."
}


// Auth API K8S

variable "username" {
  type = "string"
  description = "The username to use for HTTP basic authentication when accessing the Kubernetes master endpoint"
}

variable "password" {
  type = "string"
  description = "The password to use for HTTP basic authentication when accessing the Kubernetes master endpoint"
}


// Auto-scaling

variable "http_load_balancing_disable" {
  type = "string"
  description = "The status of the HTTP Load Balancing add-on"
}

variable "horizontal_pod_autoscaling_disable" {
  type = "string"
  description = "The status of the Horizontal Pod Autoscaling addon"
}


// Network

variable "network" {
  type = "string"
  description = "The name or self_link of the Google Compute Engine network to which the cluster is connected"
}

variable "subnetwork" {
  type = "string"
  description = "Name of the subnet to which to attach the cluster"
}


// Nodes

variable "machine_type_node" {
  type = "string"
  description = "The name of a Google Compute Engine machine type"
}

variable "disk_size_gb_node" {
  type = "string"
  description = "Size of the disk attached to each node, specified in GB"
}

variable "local_ssd_count_node" {
  type = "string"
  description = "The amount of local SSD disks that will be attached to each node pool"
}


// Node Autoscale

variable "minNodeCount_node" {
  type = "string"
  description = "Minimum number of nodes in the NodePool"
}

variable "maxNodeCount_node" {
  type = "string"
  description = "Maximum number of nodes in the NodePool"
}
```

### Debug Output
https://gist.github.com/hamshif/2307fdae8cbad4ba3195e61abb3837c3.js

### Expected Behavior
If the terraform declarations worked.
When I delete .terraform file and other generated files 
and run
```
terraform init
terraform plan
terraform apply
```
The deployment should work

### Actual Behavior
The deployment failed with timeout
Error: Error applying plan:

1 error(s) occurred:

* google_container_cluster.primary: 1 error(s) occurred:

* google_container_cluster.primary: Post 
https://container.googleapis.com/v1/projects/...-gcp-poc/zones/europe-
west1-d/clusters?alt=json: dial tcp: i/o timeout

### Steps to Reproduce
```
terraform init
terraform plan
terraform apply
```

### Important Factoids
I downgraded to google plugin 1.4 by 
1. Deleting .terraform directory 
2. Deleting .terraform.tfstate file
3. Deleting terraform.tfstate.backup file
4. Copying .terraform file from another location
```
terraform -v

Terraform v0.11.0
+ provider.google v1.4.0

Your version of Terraform is out of date! The latest version
is 0.11.2. You can update by downloading from www.terraform.io/downloads.html

```
I tried:

```
terraform init
terraform plan
terraform apply
```
https://gist.github.com/hamshif/b2c655349cb23aebe9241a2386d43d06.js

then when I ran 

```
terraform apply
```
the deployment completed successfully.

Note that I have good connectivity!!!

### References
https://stackoverflow.com/questions/48459768/why-has-terraform-stopped-working-on-gcp-project

**Comment by danawillow** [comment]
Hey @hamshif, can you try posting your debug logs in a different format? The gist links aren't working for me.

**Comment by hamshif** [comment]
Sure thanks

**Comment by hamshif** [comment]
Error: Error applying plan:

1 error(s) occurred:

* google_container_cluster.primary: 1 error(s) occurred:

* google_container_cluster.primary: Post 
https://container.googleapis.com/v1/projects/...-gcp-poc/zones/europe-
west1-d/clusters?alt=json: dial tcp: i/o timeout

**Comment by hamshif** [comment]
Error: Error applying plan:

1 error(s) occurred:

* google_container_cluster.rtp_container_cluster: 1 error(s) occurred:

* google_container_cluster.rtp_container_cluster: Error reading 
instance group manager returned as an instance group URL: Get 
https://www.googleapis.com/compute/v1/projects/rtp-gcp-
poc/zones/europe-west1-b/instanceGroupManagers/gke-rtp-container-
cluste-default-pool-8bb9aa85-grp?alt=json: dial tcp: i/o timeout

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.

**Comment by danawillow** [comment]
Hey @hamshif, would you mind posting debug logs? You can do so by running `TF_LOG=DEBUG terraform plan > debug.txt` and then pasting the output into a gist.

**Comment by hamshif** [comment]
Thanks !!!

I will also post this in the issue

<script src="
https://gist.github.com/hamshif/2d86c2dc842be8ae58f37ea69883b9f9.js
"></script>

https://gist.github.com/hamshif/2d86c2dc842be8ae58f37ea69883b9f9.js

**Comment by hamshif** [comment]
https://gist.github.com/hamshif/2d86c2dc842be8ae58f37ea69883b9f9

don't use the .js end

**Comment by danawillow** [comment]
Sorry to keep making you do this, can you run this command for me and paste the output in a gist?
```
TF_LOG=DEBUG terraform apply 2> debug.txt
```

**Comment by hamshif** [comment]
Sure no problem

**Comment by hamshif** [comment]
https://gist.github.com/hamshif/f0ca4ad95b9c1f71b620a92c7dc10f15

**Comment by hamshif** [comment]
I'm still able to provision with 1.4 plugin

**Comment by danawillow** [comment]
Looks like the error those logs are talking about is that the resource already exists. Can you try changing the name and posting new logs once you have an error output that matches the problem from your original post?

**Comment by hamshif** [comment]
I'll delete the current clusters and reapeat the process

**Comment by hamshif** [comment]
https://gist.github.com/hamshif/523cad7787d5a514525b7b5b1b770fb3

**Comment by danawillow** [comment]
Hey @hamshif, I'm totally stumped. I'm using 1.5.0 and my plan output looks the same as yours, but I'm able to create the cluster just fine. As far as I can tell from the error messages, the request to create the cluster is timing out after its been sent to GCP- I'm not sure why using a different version of Terraform would make a difference at that point. Sorry I couldn't help further :-/

**Comment by hamshif** [comment]
I deleted the gcp project from the UI [it's not possible from terraform
destroy] before I ran this so how can there be a resource?

**Comment by hamshif** [comment]
I'm willing to Skype or hangouts to demo

On Jan 31, 2018 03:24, "Dana Hoffman" <notifications@github.com> wrote:

Hey @hamshif <https://github.com/hamshif>, I'm totally stumped. I'm using
1.5.0 and my plan output looks the same as yours, but I'm able to create
the cluster just fine. As far as I can tell from the error messages, the
request to create the cluster is timing out after its been sent to GCP- I'm
not sure why using a different version of Terraform would make a difference
at that point. Sorry I couldn't help further :-/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/terraform-providers/terraform-provider-google/issues/1013#issuecomment-361791974>,
or mute the thread

**Comment by hamshif** [comment]
Hello Dana

My name is Gideon

If you are running the same configuration I then the only differences I can
think of are:

   1. The content of my variables
   2. My laptop

Can it be that parsing the variables changed?

Can it be that some security issue is timing connection out?

**Comment by hamshif** [comment]
Hi @danawillow 

I've managed to resolve my issue by mistake!!. 
When I migrated to a new project and created a new service account neither google plugin 1.4 nor 1.5 worked so I fiddled with the roles and gave myself project owner which enabled me to use the 1.5 plugin easily.

I think the resolution of this issue should be in creating more descriptive logs i.e. relaying verbose gcloud responses as part of terraform logs.

Thank you for your effort!!!
